messaging/src/main/java/org/axonframework/messaging/annotation/MessageHandlerInterceptorDefinition.java

```diff
@@ -0,0 +1,123 @@
+
+        public ResultHandlingInterceptorMember(MessageHandlingMember<T> original, Class<?> expectedResultType) {
+            super(original);
+            this.expectedResultType = expectedResultType;
+            Method method = original.unwrap(Method.class).orElseThrow(() -> new AxonConfigurationException(
+                    "The @ExceptionHandler must be on method."));
+            boolean declaredInterceptorChain = Arrays.stream(method.getParameters())
+                                                     .anyMatch(p -> p.getType().equals(InterceptorChain.class));
+            if (declaredInterceptorChain) {
+                throw new AxonConfigurationException("@ExceptionHandler annotated methods should not delare an InterceptorChain parameter");
+            }
+        }
+
+        @Override
+        public boolean canHandle(Message<?> message) {
+            return ResultParameterResolverFactory.ignoringResultParameters(() -> super.canHandle(message));
+        }
+
+        @Override
+        public Object handle(Message<?> message, T target) throws Exception {
+            InterceptorChain chain = InterceptorChainParameterResolverFactory.currentInterceptorChain();
+            try {
+                return chain.proceed();
+            } catch (Exception e) {
+                if (!expectedResultType.isInstance(e)) {
+                    throw e;
+                }
+                return ResultParameterResolverFactory.callWithResult(e, () -> {
+                    if (super.canHandle(message)) {
+                        return super.handle(message, target);
+                    }
+                    throw e;
+                });
+            }
+        }
+    }
+
+    private static class InterceptedMessageHandlingMember<T> extends WrappedMessageHandlingMember<T> implements MessageHandlerInterceptorMember<T> {
+
+        private final boolean shouldInvokeInterceptorChain;
+
+        public InterceptedMessageHandlingMember(MessageHandlingMember<T> original) {
+            super(original);
+            Method method = original.unwrap(Method.class).orElseThrow(() -> new AxonConfigurationException(
+                    "The @MessageHandlerInterceptor must be on method."));
+            shouldInvokeInterceptorChain = Arrays.stream(method.getParameters())
+                                                 .noneMatch(p -> p.getType().equals(InterceptorChain.class));
+            if (shouldInvokeInterceptorChain && !Void.TYPE.equals(method.getReturnType())) {
+                throw new AxonConfigurationException("@MessageHandlerInterceptor must return void or declare " +
+                                                             "InterceptorChain parameter.");
+            }
+        }
+
+        @Override
+        public Object handle(Message<?> message, T target) throws Exception {
+            Object result = super.handle(message, target);
+            if (shouldInvokeInterceptorChain) {
```

Comment:
What if the user didn't call `InterceptorChain#proceed`, even though a parameter was added to the `@MessageHandlerInterceptor` annotated method?

Reply:
Then apparently the interceptor thought it was necessary to prevent the invocation of the handler.

Reply:
{facepalm} But of course..thanks for that.